ocaml/Makefile

```diff
@@ -0,0 +1,171 @@
+C_INCLUDES?=-Ic/include/hacl/ \
+			-Ic/include/vale/ \
+			-Ic/include/ \
+			-I../include/ # XXX: internal is used in lib/
+KARAMEL_INCLUDES?=-Ic/include/kremlin/include \
+				  -Ic/include/kremlin/kremlib/dist/minimal/
+
+OCAMLOPT=ocamlfind opt -package ctypes,ctypes.stubs -linkpkg -I lib
+OCAMLDEP=ocamlfind dep -I lib -slash
+
+OCAMLC=ocamlfind c -g -package ctypes,ctypes.stubs -linkpkg -I lib
+
+CFLAGS += -I "$(shell ocamlfind query ctypes)" -I "$(shell ocamlfind c -where)" \
+		  $(C_INCLUDES) $(KARAMEL_INCLUDES)
+
+# Don't include bindings for files that cannot be compiled.
+BLOCKLIST_ML=$(patsubst %.c,%,$(BLOCKLIST))
+ALL_OCAML=$(filter-out $(BLOCKLIST_ML),$(patsubst lib_gen/%_gen.ml,%,$(wildcard lib_gen/*_gen.ml)))
+
+# File names.
+ALL_BINDINGS=$(patsubst %,lib/%_bindings.cmx,$(ALL_OCAML))
+ALL_GENERATORS=$(patsubst %,lib_gen/%_gen.exe, $(ALL_OCAML))
+ALL_ML_STUBS=$(patsubst %,lib/%_stubs.cmx,$(ALL_OCAML))
+ALL_C_STUBS=$(patsubst %,lib/%_c_stubs.o,$(ALL_OCAML))
+
+include .depend.ocaml
+include ctypes.depend
+
+# - copy and clean config.h
+setup:
+ifeq ($(wildcard $(C_PATH)),)
+		mkdir $(C_PATH)
+		cp -r ../build/installed/* $(C_PATH)/
+		cp $(C_LIB) .
+		sed -ri '/(\/\/.*)|(\/\*.*\*\/)/d' ./c/include/hacl/config.h
+endif
+
+lib_gen/Lib_RandomBuffer_System_gen.cmx: lib/Lib_RandomBuffer_System_bindings.cmx
+lib_gen/Lib_RandomBuffer_System_gen.exe: lib/Lib_RandomBuffer_System_bindings.cmx lib_gen/Lib_RandomBuffer_System_gen.cmx
+
+.depend.ocaml:
+	$(OCAMLDEP) $(wildcard lib/*.ml) $(wildcard lib_gen/*.ml) > $@
+
+%.exe:
+
+lib_gen/%_gen.exe:
+	$(OCAMLOPT) $(filter-out %.a,$^) $(C_LIB) -o $@
+
+%.cmx: %.ml
+	$(OCAMLOPT) -c $^ -o $@
+
+%.cmo: %.ml
+	$(OCAMLC) -c $^ -o $@
+
+
+.PRECIOUS: lib/%_stubs.ml
+lib/%_stubs.ml: lib/%_c_stubs.c
+
+lib/%_stubs.ml lib/%_c_stubs.c: lib_gen/%_gen.exe
+	$<
+
+BLOCKLIST_CMX =  $(patsubst %,lib/%_stubs.cmx,$(BLOCKLIST_ML))
+BLOCKLIST_CMX += $(patsubst %,lib/%_bindings.cmx,$(BLOCKLIST_ML))
+CTYPES_CMX    =  $(filter-out $(BLOCKLIST_CMX),$(CTYPES_DEPS))
+CTYPES_CMX    += lib/Lib_RandomBuffer_System_stubs.cmx lib/Lib_RandomBuffer_System_bindings.cmx
+CTYPES_ML     =  $(patsubst %.cmx,%.ml,$(CTYPES_CMX))
+CTYPES_CMI    =  $(patsubst %.cmx,%.cmi,$(CTYPES_CMX))
+CTYPES_CMO    = $(patsubst %.cmx,%.cmo,$(CTYPES_CMX))
+
+
+ocamlevercrypt.cma: $(ALL_BINDINGS) $(CTYPES_CMO) $(ALL_C_STUBS) $(CTYPES_CMX)
+	ocamlmklib -o ocamlevercrypt $(CTYPES_CMO) -L. -l$(STATIC_C_LIB_NAME)
+
+ocamlevercrypt.cmxa: $(ALL_BINDINGS) $(ALL_ML_STUBS) $(ALL_C_STUBS)
+	ocamlmklib -o ocamlevercrypt $(CTYPES_CMX) -L. -l$(STATIC_C_LIB_NAME)
+
+STUBLIBS_PATH=$(OPAM_SWITCH_PREFIX)/lib/stublibs
+
+dllocamlevercrypt.$(OCAML_SO): ocamlevercrypt.cmxa ocamlevercrypt.cma
+	ocamlmklib -o ocamlevercrypt $(ALL_C_STUBS) -L. -L$(STUBLIBS_PATH) -l$(STATIC_C_LIB_NAME)
+
+clean:
+	rm -rf *.$(SO) *.$(OCAML_SO) *.a .depend.ocaml *.cmxa *.cma \
+	**/*.cma **/*.cmo **/*.cmx **/*.cmxa* **/*.o **/*.c **/*.cmi **/*.exe **/*.o **/*.d \
```

Comment:
you are relying on the shell to perform wildcard expansion... are you using zsh? this doesn't work by default on bash; I would suggest using $(wildcard), or $(shell find) with POSIX options

Reply:
I'd rather remove this than getting more into make weirdness 🤪 It hasn't been there before.
I always forget that bash still can't do this 😔